Java/Graphs/Hamiltonian_path.java

```diff
@@ -0,0 +1,101 @@
+import java.util.Scanner;
+import java.util.Arrays;
+ 
+
+public class Hamiltonian_path
+{
+    private int V, count;
+    private int[] path;     
+    private int[][] graph;
+ 
+    /** Function to find cycle **/
+    public void findHamiltonianCycle(int[][] g)
+    {
+        V = g.length;
+        path = new int[V];
+ 
+        Arrays.fill(path, -1);
+        graph = g;        
```

Comment:
Try to add some comments here expressing what you are actually doing !

Reply:
Done